reference_impl/reference_impl.rs

```diff
@@ -382,3 +380,9 @@ impl Hasher {
         output.root_output_bytes(out_slice);
     }
 }
+
+impl Default for Hasher {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
One of the reasons I've leaned against publishing this implementation on crates.io is that there's a tension between keeping it simple and making it actually usable. In addition to `Default`, a real `Hasher` should also implement `Debug` and `Clone` and probably `Write`. But on the other hand, the primary goal of `reference_impl.rs` is to be readable, including to people who don't actually know Rust. I think adding common traits into the mix detracts from that, because less experienced readers don't know what `#[derive(...)]` means, and it's not clear that they can gloss over it.